src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -97,9 +98,24 @@ private void getMealList(HttpServletRequest request, HttpServletResponse respons
         // Every object Meal adds if it contains at least one keyword in at least one field 
         // (type, title, description, ingredients).
         List<Meal> searchedMeal = new ArrayList<>();
+        HashMap<Integer, List<Meal>> sortedMeal = new HashMap<>();
         for (Meal meal : meals) {
-            if (isResultOfSearch(meal, params)) {
-                searchedMeal.add(meal);
+            int frequency = getFrequency(meal, params);
```

Comment:
Add a test that checks whether an entry with more matches is returned at the beginning of the results list, favoured over other entries.